src/systems/waypoint-system.js

```diff
@@ -365,10 +365,14 @@ export class WaypointSystem {
       ) {
         elementFromTemplate.object3D.visible = this.scene.is("frozen");
         if (elementFromTemplate.object3D.visible) {
-          this.viewingCamera = this.viewingCamera || document.getElementById("viewing-camera").object3D;
+          this.viewingCamera = this.viewingCamera || document.getElementById("viewing-camera").object3DMap.camera;
           setMatrixWorld(
             elementFromTemplate.object3D,
-            calculateIconTransform(waypointComponent.el.object3D, this.viewingCamera, this.helperMat4)
+            calculateIconTransform(
+              waypointComponent.el.object3D,
+              this.viewingCamera.object3DMap.camera,
```

Comment:
This `object3DMap.camera` seems redundant with the `object3DMap.camera` above

Reply:
:+1: nice catch. I'll make sure to test the features I've touched as well.